src/migration-scripts/isis/1-to-2

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# T4739 refactor, and remove "on" from segment routing from the configuration
+
+from sys import argv
+from sys import exit
+
+from vyos.configtree import ConfigTree
+
+if (len(argv) < 1):
+    print("Must specify file name!")
+    exit(1)
+
+file_name = argv[1]
+
+with open(file_name, 'r') as f:
+    config_file = f.read()
+
+config = ConfigTree(config_file)
+
+# Check if ISIS segment routing is configured. Then check if segment routing "on" exists, then delete the "on" as it is no longer needed. This is for global configuration.
+if config.exists(['protocols', 'isis']):
```

Comment:
min/max values of high/low label values got adjusted, too. As I am not an expert, but could they be moved from e.g.

* 16 -> 116
* 18 -> 118?

or can we set every value < 100 to 100?

Reply:
Sure we can make them anything we want. So generally in MPLS we can go from 16-1048576. Below 16 is reserved MPLS labels. I did some testing for OSPF and ISIS segment routing inside of FRR and I found out that ISIS and OSPF reserve MPLS label ranges differently. It seems ISIS is able to reserve lower, whereas OSPF was not. I found out specifically for OSPF that 80 was the minimum label value we could reserve. I haven't gotten an answer as to yet of why this is but the current minimum value was 80. I chose to give a little bit extra worth of room for both ISIS and OSPF for FRR in case the maintainers want to do some more reservations for label ranges and whatnot. That is why I bumped the minimum label value to 100. I was able to test this to work for both ISIS and OSPF and they both worked properly in adding the configuration to FRR. I also made the smoketests do this minimum label check as well so that in case there's an FRR change we'll immediately see it fail on smoke tests.

> or can we set every value < 100 to 100?

We can set the minimum MPLS label value to 100. We can set it lower, but due to the inconsistency between how ISIS and OSPF reserve labels through segment routing I decided to make 100 the minimum so that we don't run into those inconsistencies anymore. We are also not losing too much label range either. We can go from 100-1048576 which should be absolutely plenty.

Reply:
Forgot to add, if you're curious you can see the debugs I did here:

https://github.com/FRRouting/frr/issues/12007